Standalone Sensor Mode - Make creport.json file list complete

iximiuz opened the pull request:
While a proper solution is still required, at the moment it's done by re-enumerating the artifacts/files folder right before generating the creport file.

iximiuz committed:
Standalone Sensor Mode - Make creport.json file list complete

While a proper solution is still required, at the moment it's done by
re-enumerating the artifacts/files folder right before generating the
creport file.

kcq commented on pkg/app/sensor/app.go:
```diff
@@ -29,6 +29,10 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+const (
```
might be good to have global consts for the artifact related directory names and files, so we can share them in multiple places instead of redefining them

iximiuz replied:
By _global consts_ you mean a separate file/package? The `defaultArtifactDirName` was defined in `artifcats.go` but not really used there. So, I figured moving it closer to the place it's actually used might be a good idea. But it's not really a global constant - it's more like a default value of a (to be introduced) flag.

Overall, though, I agree with the idea :)

iximiuz committed:
Fix download link in the version info. (#391)

mritunjaysharma394 committed:
feat: enhance json console output in xray and version (#384)

* initial step for json console output in xray and version

Signed-off-by: Mritunjay Sharma <mritunjaysharma394@gmail.com>

* refactors xray json

Signed-off-by: Mritunjay Sharma <mritunjaysharma394@gmail.com>

* enhances json support for version

Signed-off-by: Mritunjay Sharma <mritunjaysharma394@gmail.com>

* refactor prefix cmd

Signed-off-by: Mritunjay Sharma <mritunjaysharma394@gmail.com>

Signed-off-by: Mritunjay Sharma <mritunjaysharma394@gmail.com>

dependabot[bot] committed:
Bump test/bats from `c97b3a1` to `ac68107` (#388)

Bumps [test/bats](https://github.com/bats-core/bats-core) from `c97b3a1` to `ac68107`.
- [Release notes](https://github.com/bats-core/bats-core/releases)
- [Commits](https://github.com/bats-core/bats-core/compare/c97b3a1b8644ddede18bff6bc035c12e9f50be78...ac68107a625c7294e5098a0f218ab85da4e215ef)

---
updated-dependencies:
- dependency-name: test/bats
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump test/bats from `ac68107` to `0e8d70b` (#392)

Bumps [test/bats](https://github.com/bats-core/bats-core) from `ac68107` to `0e8d70b`.
- [Release notes](https://github.com/bats-core/bats-core/releases)
- [Commits](https://github.com/bats-core/bats-core/compare/ac68107a625c7294e5098a0f218ab85da4e215ef...0e8d70b4faca091e34076b14ccc8be28c21f1b65)

---
updated-dependencies:
- dependency-name: test/bats
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

iximiuz committed:
feat: Standalone Sensor Mode (#383)

Sensor got two new flags (the exact names to be refined thouhg):

-mode=controlled|standalone
-command=<JSON-encoded MonitorStart command>

The default mode is 'controlled' and sensor's behavior in that mode
(supposedly) stayed unchanged.

In the 'standalone' mode, sensor becomes independent from the driver app
(docker-slim). Sensor terminates as long as the target app is terminated
and the artifacts have been persisted.

Additionally, the signal forwarding (from the sensor's process to the
target app) is implemented, including sending SIGKILL to the target app
after a grace period. Beware, though, that the signal forwarding works
only in the standalone mode.
